Libraries/libbom/Sources/bom.c

```diff
@@ -245,9 +255,16 @@ bom_index_add(struct bom_context *context, const void *data, size_t data_len)
     struct bom_index_header *index_header = (struct bom_index_header *)((void *)header + ntohl(header->index_offset));
 
     /* Insert index at the end of the list. */
+    bool resize = true;
+    if (context->preallocated_index_count > 0) {
```

Comment:
preallocated_index_count is the optimization, but I'll remove it anyway so that the code can modify an existing file in the future more easily.